internal/pkg/deploy/cloudformation/stack/task.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"strconv"
+
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+const (
+	taskTemplatePath = "task/cf.yml"
+
+	taskNameParamKey         = "TaskName"
+	taskCPUParamKey          = "TaskCPU"
+	taskMemoryParamKey       = "TaskMemory"
+	taskLogRetentionParamKey = "LogRetention"
+
+	taskContainerImageParamKey = "ContainerImage"
+	taskTaskRoleParamKey       = "TaskRole"
+	taskCommandParamKey        = "Command"
+
+	taskLogRetentionInDays = "1"
+)
+
+type taskStackConfig struct {
```

Comment:
not really important for this, just more to keep in mind for the future so that everyone can see: but I think it'd be cool if we rename the structs to be just `stack.task` vs. `stack.taskStackConfig`. In this situation it doesn't matter much since the struct is private.

and then the input to the constructs can be `stack.TaskConfig` instead of being in the `deploy` pkg.